### PR TITLE
Fix some pylint issues and clean up file handles where appropriate

### PR DESCRIPTION
**Pull request by nonowazu**
**Purpose**
More fixes for pylint and closing file handles when the parent object goes through garbage collection

**New Features**
Less pylint bugs, less potential memory leaks.

### CODE REVIEW

**Review by lunarrize — APPROVED** [review]

File: nari/io/writer/pickle.py [hunk 1]
```diff
@@ -10,7 +10,10 @@ class PickleWriter(Writer):
     """Writes a stream of events into a binary file"""
     def __init__(self, stream: Iterator[Event], filename: str):
         super().__init__(stream)
-        self.handle = open(filename, 'wb')
+        self.handle = open(filename, 'wb') # pylint: disable=consider-using-with
+
+    def __del__(self):
+        self.handle.close()
```
Comment (ayyaruq) [inline nari/io/writer/pickle.py:16]: I see the docstring in the reader but looks like you missed it here.

File: nari/io/reader/actlog.py [hunk 2]
```diff
@@ -10,11 +10,12 @@ class ActLogReader(Reader):
     """Implementation of the Reader class for parsing ACT network logs"""
     def __init__(self, actlog_path, *, raise_on_checksum_failure=False, raise_on_invalid_id=False):
         # might need a helper function, but eh.
-        self.handle = open(actlog_path, 'r')
+        self.handle = open(actlog_path, 'r') # pylint: disable=consider-using-with
         # Let's talk about this. ACT network logs seem to use two different indexes – one for
         # network events, and one for other types of events. Without breaking the code too badly,
         # might be good to identify if an ID is based off network or 'other' (memory events only?)
         # and cycle through the correct indexes
+        # TODO:
```
Comment (ayyaruq) [inline nari/io/reader/actlog.py:18]: Empty TODO?

File: nari/io/writer/pickle.py (hunk 1 above)
Comment (nonowazu) [reply nari/io/writer/pickle.py]: should be good to go now.

File: nari/io/reader/actlog.py (hunk 2 above)
Comment (nonowazu) [reply nari/io/reader/actlog.py]: I was adding it in reaction to the comment above, but I added more 'actionable' text in the TODO

**Review by ayyaruq — APPROVED** [review]

**Pull request merged**